iree/turbine/kernel/wave/cache.py

```diff
@@ -70,6 +72,7 @@ def extract_arg_types(kernel_fn: Callable):
     return [
         (arg.annotation, arg.annotation.physical_layout)
         for arg in inspect.signature(kernel_fn).parameters.values()
+        if not isinstance(arg.annotation, DataType)
```

Comment:
can we generalize this some more, can you check what is the expected arg.annotation type and check for `if isinstance(arg.annotation, <TargetType>)`

Reply:
Done. Checking for `KernelBufferMeta`